examples/medplum-provider/src/pages/spaces/space-persistence.ts

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: Copyright Orangebot, Inc. and Medplum contributors
+// SPDX-License-Identifier: Apache-2.0
+import type { Communication } from '@medplum/fhirtypes';
+import type { MedplumClient } from '@medplum/core';
+import type { Message } from '../../types/spaces';
+
+/**
+ * Creates a new conversation topic (main Communication resource)
+ * @param medplum - The Medplum client instance
+ * @param title - The title of the conversation
+ * @param model - The AI model being used
+ * @returns The created Communication resource
+ */
+export async function createConversationTopic(
+  medplum: MedplumClient,
+  title: string,
+  model: string
+): Promise<Communication> {
+  const profile = medplum.getProfile();
+  if (!profile?.id) {
+    throw new Error('Profile not found');
+  }
+
+  const topic: Communication = {
+    resourceType: 'Communication',
+    status: 'in-progress',
+    identifier: [
+      {
+        system: 'http://medplum.com/ai-message',
+        value: 'ai-message-topic',
+      },
+    ],
+    subject: {
+      reference: `Practitioner/${medplum.getProfile()?.id}`,
+    },
+    topic: {
+      text: title,
+    },
+    note: [
+      {
+        text: JSON.stringify({ model }),
+      },
+    ],
+  };
+
+  return medplum.createResource(topic);
+}
+
+/**
+ * Saves a message as a Communication resource linked to the topic
+ * @param medplum - The Medplum client instance
+ * @param topicId - The ID of the conversation topic
+ * @param message - The message to save
+ * @param sequenceNumber - The sequence number of the message
+ * @returns The created Communication resource
+ */
+export async function saveMessage(
+  medplum: MedplumClient,
+  topicId: string,
+  message: Message,
+  sequenceNumber: number
+): Promise<Communication> {
+  const communication: Communication = {
+    resourceType: 'Communication',
+    status: 'completed',
+    identifier: [
+      {
+        system: 'http://medplum.com/ai-message',
+        value: 'ai-message',
+      },
+    ],
+    partOf: [
+      {
+        reference: `Communication/${topicId}`,
+      },
+    ],
+    payload: [
+      {
+        contentString: JSON.stringify({
```

Comment:
I don't know what shape the data being saved here is, but any risk of it causing the `Communication` to come close to the 1MB limit?